src/solvers/prop/prop_conv.h

```diff
@@ -110,14 +110,14 @@ class prop_conv_solvert:public prop_convt
 
   virtual void clear_cache() { cache.clear();}
 
-  typedef std::map<irep_idt, literalt> symbolst;
-  typedef std::unordered_map<exprt, literalt, irep_hash> cachet;
```

Comment:
Please don't do this. Type names also carry useful information about what a data structure is intended to hold/do.

Reply:
`typedef`s do the opposite. Especially when concealing standard data structure, such as `map`. When reading code using typedefs this way one might think that `cachet` and `symbolst` is a user-defined class that needs special treatment. When explicitly written as `std:::vector` or `std::map` the interface and structure of the variable becomes obvious. The name of the variable should signify intent, not the type alias.

Reply:
As you please.

Reply:
`symbolst` is not great here, but `cachet` actually conveys intent.
Unless the typedef is only used once, I'm strongly in favour of using typedefs, in particular at interfaces.
If you think that typedefs hide information then you should use an IDE that solves that problem for you.

Reply:
Yeah; even if used once, these can still be worth having.

Reply:
I think `symbolst` is misleading (you would expect a vector of `symbolt`). `cachet` does carry the intent but it's already present in the variable name (`cache`).